File: sum_of_special_numbers.rb
def checkNumber(number)
  sum = []
 for i in 2..number
    if is_prime(i) == false && is_divisible_by_six(i) == false
      sum << i
      puts i
    end
  end
  result = sum.map(&:to_i).reduce(0, :+)
  result
end

def is_prime(num)
  (2..(num - 1)).each do |n|
    return false if num % n == 0
  end
  true
end

def is_divisible_by_six(num)
  return true if num % 6 == 0
  false
end

File: sum_of_special_numbers/sum_of_special_numbers.py
def checkNumber(number):
  numbers = []
  for i in range(2, int(number)+1):
    if is_prime(i) == False and is_divisible_by_six(i) == False:
      numbers.append(i)
  result = sum(numbers)
  return(result)

def is_prime(num):
  for n in range(2, int(num**1/2)+1):
    if num % n == 0:
      return False 
  return True


def is_divisible_by_six(num):
  if num % 6 == 0: 
    return True
  return False

print checkNumber(12)
